pandas/core/internals/managers.py

```diff
@@ -403,6 +404,43 @@ def reduce(
             new_mgr = type(self).from_blocks(res_blocks, [self.items, index])
         return new_mgr, indexer
 
+    def grouped_reduce(
+        self: T, func: Callable, ignore_failures: bool = False
+    ) -> Tuple[T, np.ndarray]:
+        """
+        Apply grouped reduction function blockwise, returning a new BlockManager.
+
+        Parameters
+        ----------
+        func : grouped reduction function
+        ignore_failures : bool, default False
+            Whether to drop blocks where func raises TypeError.
+
+        Returns
+        -------
+        BlockManager
+        """
+        result_blocks: List[Block] = []
+
+        for blk in self.blocks:
+            try:
+                applied = blk.apply(func)
+            except (TypeError, NotImplementedError):
+                if not ignore_failures:
+                    raise
+                continue
+            result_blocks = extend_blocks(applied, result_blocks)
+
+        if len(result_blocks) == 0:
+            raise DataError("No numeric types to aggregate")
+
+        index = Index(range(result_blocks[0].values.shape[-1]))
```

Comment:
can we figure out this length before calling grouped_reduce?  if so, i think that we can turn it into an argument, then refactor reduce, apply and possibly quantile to dispatch to this

Reply:
The length can certainly be known before hand for the groupby case (`self.grouper.ngroups`), I am only not fully sure this is a nice API for the Manager